src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

```diff
@@ -1479,13 +1479,13 @@ services:
     tags:
       - { name: form.type }
 
-  form.type.sell.product.details.feature_value_type:
-    class: 'PrestaShopBundle\Form\Admin\Sell\Product\Details\FeatureValueType'
+  PrestaShopBundle\Form\Admin\Sell\Product\Details\FeatureValueType:
     parent: 'form.type.translatable.aware'
+    autowire: true
+    public: false
     arguments:
-      - '@prestashop.adapter.form.choice_provider.features_choice_provider'
-      - '@form.type.sell.product.event_listener.feature_value_listener'
-    public: true
+      $featuresChoiceProvider: '@prestashop.adapter.form.choice_provider.features_choice_provider'
+      $featureValueListener: '@PrestaShopBundle\Form\Admin\Sell\Product\EventListener\FeatureValueListener'
```

Comment:
Why not using auto wiring ?

Reply:
it depends on interface, so we must define which actual class to inject